### DISCUSSION TITLE
Problem during the economical optimization

### DISCUSSION THREAD

**Discussion by Vinxenx** [SESMG/SESMG#118] [Q&A] [answered]
In meiner Simulation möchte ich ein Quartier darstellen, dass an die Netze angeschlossen ist und zur Wärmeerzeugung einen Heizkessel und Wärmespeicher nutzen soll. Die Ergebnisse verwirren mich jedoch:
- Warum wird das maximale Investment bei den thermischen Speichern verbaut, obwohl diese nicht einmal genutzt werden?
- Warum werden die periodischen Kosten im Feld constraints/CU angezeigt (constraints cost sind in der gesamten Tabelle bei null, lediglich periodical costs wurden integriert für die verbauten Technologien) 
- Nachdem mir als Fehler "sequences" angezeigt wurde, habe ich einige work-a-rounds ausprobiert. Nachdem ich  die Simulation um X_heat_bus erweitert hatte, hat sie  funktioniert hat. Diese werden in den Ergebnissen aber nicht einmal genutzt.

Vielen Dank im Voraus. 

[1.2.1 Generisches Quartier konventionell mit Gaskessel und Wärmespeicher.xlsx](https://github.com/chrklemm/SESMG/files/9283603/1.2.1.Generisches.Quartier.konventionell.mit.Gaskessel.und.Warmespeicher.xlsx)

**Comment by GregorBecker** [comment, marked as answer]
Good evening @Vinxenx, 

I just optimized your scenario with the current master branch. The only costs arising within the optimization of your scenario are variable costs, since the simulation shows that a pure shortage supply is the cheapest opportunity. 

The operation of the district heating, i.e. your heatshortages, appears very cheap on the first consideration, which has the consequence that the gas heating as well as the thermal storage are uneconomical.

Have you used another branch for your calculations? 

To your questions: 

1. I could not reproduce that the thermal storages are designed. In my results the capacities are 0.

2. I could not reproduce that either, in my case there are no emissions. 

3. now that I have also disabled the heat_bus_shortages, the gas heaters are running. Further the storages are no investment alternative and remain unconsidered.

Kind regards, Gregor

**Reply by Vinxenx** [reply]
Thank u for the fast answer! I willl check it directly

Best regards, Vincent

**Reply by Vinxenx** [reply]
I really don't know, if I have the newest branch. Because i have some issues with installing SESMG, I'm scared to change soemthing. To your answer: 
This is interesting. Like u said it make sense, that the simulation use the heat_bus as cheapest option. This is why i dont understand, that my solver isnt't showing me the variable cost of the shortages. Every cost is displayed under constraint costs, like in the .png.  I usually expect, that the cheapest option are dispayed in the dashboard under total system cost, but not under constraints/CU. 


![image](https://user-images.githubusercontent.com/96150761/183519455-dc2f7000-917d-4df0-bf02-dd829d541dda.png)

**Reply by jtock** [reply]
Hi Vinxenx, 
we can not understand your results in view of your input parameters. Can you send us a screenshot of your GUI settings? It is possible that you chose some wrong settings. 
Greetings, Jan

**Reply by Vinxenx** [reply]
This is definitly possible! 

![image](https://user-images.githubusercontent.com/96150761/183615019-92b6bf0e-cffd-405a-9ac2-c0d76b300040.png)

**Reply by jtock** [reply]
You need to deactivate "Switch Criteria". You will probably generate correct results afterwards.

**Reply by Vinxenx** [reply]
Thank you so much! Now, everything make sense.

Kind regards, vincent
